sqld/src/replication/primary/logger.rs

```diff
@@ -236,51 +262,98 @@ impl LogFile {
         &self.header
     }
 
-    pub fn write_header(&mut self, header: &LogFileHeader) -> anyhow::Result<()> {
-        self.file.write_all_at(bytes_of(header), 0)?;
+    pub fn commit(&mut self) -> anyhow::Result<()> {
+        self.header.frame_count += self.uncommited_frame_count;
+        self.uncommited_frame_count = 0;
+        self.commited_checksum = self.uncommited_checksum;
+        self.write_header()?;
+
+        Ok(())
+    }
+
+    fn rollback(&mut self) {
+        self.uncommited_frame_count = 0;
+        self.uncommited_checksum = self.commited_checksum;
+    }
```

Comment:
Could we try to truncate the log file, so that the rolled-back log entries don't consume disk space? This can be meaningful for large transactions.

Reply:
we could, but at the same time, it makes the rollback fallible. My assumption is that there will be other writes that will overwrite trash data, and at some point the log will be compacted

Reply:
We can truncate the file on a "best effort" basis: if this operation fails, we can just ignore it.

Reply:
https://github.com/libsql/sqld/issues/428 issue opened